From 66028e3220966ec4337688f7eee5f0a5ca27c149 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sat, 26 Apr 2025 15:30:37 +0200
Subject: [PATCH 08/55] target: don't use GENERIC_PLATFORM_DIR for generic
 other-files

Generic other-files directory path hardly needs to be overwritten as 99%
of the time the mainline generic one is usable.

Don't depend on GENERIC_PLATFORM_DIR variable (that can point to the
external target feed path) but introduce a dedicated define and set it
to the default TOPLEVEL location by default.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 include/kernel-defaults.mk | 2 +-
 include/target.mk          | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/kernel-defaults.mk b/include/kernel-defaults.mk
index cbe415abe6..372da1b085 100644
--- a/include/kernel-defaults.mk
+++ b/include/kernel-defaults.mk
@@ -187,7 +187,7 @@ define Kernel/CompileImage/Initramfs
 	$(call locked,{ \
 		$(if $(2),$(call Kernel/PrepareConfigPerRootfs,$(LINUX_DIR)$(2));) \
 		$(call Kernel/Configure/Initramfs,$(if $(1),$(1),$(TARGET_DIR)),$(LINUX_DIR)$(2)); \
-		$(CP) $(GENERIC_PLATFORM_DIR)/other-files/init $(if $(1),$(1),$(TARGET_DIR))/init; \
+		$(CP) $(GENERIC_OTHER_FILES_DIR)/init $(if $(1),$(1),$(TARGET_DIR))/init; \
 		$(if $(SOURCE_DATE_EPOCH),touch -hcd "@$(SOURCE_DATE_EPOCH)" $(if $(1),$(1),$(TARGET_DIR)) $(if $(1),$(1),$(TARGET_DIR))/init;) \
 		rm -rf $(LINUX_DIR)$(2)/usr/initramfs_data.cpio*; \
 		$(if $(CONFIG_TARGET_ROOTFS_INITRAMFS_SEPARATE), \
diff --git a/include/target.mk b/include/target.mk
index efaf8cb0bd..637ef2077e 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -177,6 +177,7 @@ GENERIC_BACKPORT_DIR := $(GENERIC_PLATFORM_DIR)/backport$(if $(wildcard $(GENERI
 GENERIC_PATCH_DIR := $(GENERIC_PLATFORM_DIR)/pending$(if $(wildcard $(GENERIC_PLATFORM_DIR)/pending-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
 GENERIC_HACK_DIR := $(GENERIC_PLATFORM_DIR)/hack$(if $(wildcard $(GENERIC_PLATFORM_DIR)/hack-$(KERNEL_PATCHVER)),-$(KERNEL_PATCHVER))
 GENERIC_FILES_DIR := $(foreach dir,$(wildcard $(GENERIC_PLATFORM_DIR)/files $(GENERIC_PLATFORM_DIR)/files-$(KERNEL_PATCHVER)),"$(dir)")
+GENERIC_OTHER_FILES_DIR := $(TOPDIR)/target/linux/generic/other-files
 
 __config_name_list = $(1)/config-$(KERNEL_PATCHVER) $(1)/config-default
 __config_list = $(firstword $(wildcard $(call __config_name_list,$(1))))
-- 
2.34.1

